gqcp/tests/ONVBasis/OperatorString_test.cpp

```diff
@@ -0,0 +1,177 @@
+    const auto operator_string_indices = operator_string.operatorIndices();
+
+    // Check whether the indices are stored correctly.
+    BOOST_CHECK_EQUAL_COLLECTIONS(indices.begin(), indices.end(), operator_string_indices.begin(), operator_string_indices.end());
+}
+
+/**
+ *  Check whether the SpinUnresolvedOperatorString`s `isZero()` check correctly checks whether the operator string will inherently equal zero when applied to any ONV.
+ */
+BOOST_AUTO_TEST_CASE(test_unresolved_isZero) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    // One vector does contain duplicate indices, one does not.
+    std::vector<size_t> indices_1 = {1, 4, 7, 8};
+    std::vector<size_t> indices_2 = {1, 4, 7, 1};
+
+
+    // Initialize the SpinUnresolvedOperatorStrings corresponding to the indices.
+    const auto operator_string_1 = GQCP::SpinUnresolvedOperatorString {indices_1};
+    const auto operator_string_2 = GQCP::SpinUnresolvedOperatorString {indices_2};
+
+    // Use the `isZero()` API to check whether the operator string will always result in a zero value.
+    const auto operator_string_1_zero_check = operator_string_1.isZero();
+    const auto operator_string_2_zero_check = operator_string_2.isZero();
+
+    // Compare the result of the `isZero()` check with the expected boolean value (false for the operator string without duplicate indices, true for the one with duplicate indices).
+    BOOST_CHECK_EQUAL(operator_string_1_zero_check, false);
+    BOOST_CHECK_EQUAL(operator_string_2_zero_check, true);
+}
+
+/**
+ *  Check whether the SpinResolvedOperatorString stores its information correctly.
+ */
+BOOST_AUTO_TEST_CASE(test_resolved_indices_spins) {
+
+    // Initialize the vector containing the indices we want to use for our operator string.
+    std::vector<size_t> indices = {1, 4, 7, 8};
+
+    // Initialize the vector containing the spin components we want to use for our operator string indices.
+    std::vector<GQCP::Spin> spins = {GQCP::Spin::alpha, GQCP::Spin::beta, GQCP::Spin::alpha, GQCP::Spin::beta};
+
+    // Initialize the SpinUnresolvedOperatorString.
+    const auto operator_string = GQCP::SpinResolvedOperatorString(indices, spins);
+
+    // Get the indices and spins from the operator string.
+    const auto operator_string_indices = operator_string.operatorIndices();
+    const auto operator_string_spins = operator_string.operatorSpins();
+
+    // Check whether the indices and spins are stored correctly.
+    BOOST_CHECK_EQUAL_COLLECTIONS(indices.begin(), indices.end(), operator_string_indices.begin(), operator_string_indices.end());
+    BOOST_CHECK_EQUAL_COLLECTIONS(spins.begin(), spins.end(), operator_string_spins.begin(), operator_string_spins.end());
+}
+
+/**
+ *  Check whether the SpinResolvedOperatorString can be correctly `SpinResolved` in its alpha and beta component, with the correct phase factors.
+ */
+BOOST_AUTO_TEST_CASE(test_spin_resolve) {
```

Comment:
I don't think this test is correct. I expect `(0a 1b 2a 3b 4a)` to yield a phase factor of `(-1)^3 = -1` because I count 3 beta operators before all the alpha operators:
- `0a` has no beta operators before it, so it is already in the correct place
- `2a` has 1 beta operator before it, so we need 1 anti-commutation: `p=-1`
- `4a` has 2 beta operators before it, so we need 2 anti-commutations: `p=(-1)^2`

So in total, we have `p=(-1)^3 = -1`.

Could you add **two** other tests that you work out by hand to check the implementation?

Reply:
Indeed, this is due to the error I made concerning the vacuum. 

While implementing, I printed the intermediate phase factor values and this did result in what you are describing here: `(-1)^3`. The `p=1` I was testing for comes from the factor `^2` I added for moving the vacuum. I`ll remove the power and add additional tests to support this.